### ISSUE TITLE
Error in comments writing `type:`

### ISSUE THREAD

**Issue by hrmorley34** [PyCQA/pylint#4028]
### Steps to reproduce
1. Create a file containing a comment beginning `type:`, such as
```py
# type text
# type: text <- error here
# dont type: text <- this is fine
# type : nothing <- this is also fine
```
2. Run pylint on it (either through VSCode or using `py.exe -m pylint`)

### Current behavior
```
test.py:2:9: E0001: invalid syntax (<unknown>, line 2) (syntax-error)
```
It errors on anything with `#` followed by some space followed by `type:`.

### Expected behavior
No error, because it is in a comment.

### pylint --version output
```
pylint 2.6.0
astroid 2.4.2
Python 3.9.1 (tags/v3.9.1:1e5d33e, Dec  7 2020, 17:08:21) [MSC v.1927 64 bit (AMD64)]
```

**Comment by dbaty** [comment]
I can reproduce this with the master branch on both pylint and astroid.

Here astroid (a dependency of pylint) detects a (badly written) type comment. It happens [here in astroid](https://github.com/PyCQA/astroid/blob/c9fd1934e9c49d9052f64439fc7ea82026bce00f/astroid/builder.py#L167-L170). (In fact, the error is raised by either the third-party `typed_ast` library, or the `ast` standard library, depending on the Python version.)

This issue has already been reported (and closed) here in pylint in #2299, which leads to python/typed_ast/issues/62, which itself has been closed as "won't fix" by a certain Guido van R. In other words, I suppose that it works as intended and is unlikely to be changed and fixed.
